### DISCUSSION TITLE
vector trimming

### DISCUSSION THREAD

**Discussion by mshoek** [keiwando/toonsquid#386] [Ideas]
Something like Adobe Fresco's vector trimming would be really nice to have in ToonSquid. Basically, the vector trimmer removes intersecting vector strokes.

![Fresco-June-2020-rel--Vector-trimming](https://github.com/keiwando/toonsquid/assets/118011378/1bc17c26-79e5-411a-8345-645352d0c2ec)

Chris Piascik explains it further in this video:
[![Vector trimming](https://i.postimg.cc/WzcKvnSn/chris-piascik-vector-trimming.png)](https://youtu.be/EbN6t3QJBKE "Vector trimming")

**Comment by ota-works** [comment]
OMG! THIS!!!! This would make Cleanup so efficient!!! Clip Studio Paint does this even better: It creates a vector layer where you can EVEN USE BITMAP BRUSHES AS VECTOR BRUSHES!!! So cool!!! and it adds vector points to the stroke so you can add further adjustments to the vector line with a bitmap brush.

It would be so so dope if Toonsquid could implement this in an update.

**Reply by ota-works** [reply]
https://www.youtube.com/watch?v=r3SQJAViG4Q&pp=ygUhdmVjdG9yIGxheWVyIGluIGNsaXAgc3R1ZGlvIHBhaW50

This video gives an overview of the Vector Layer feature. The reason why I want this is because I love using other brushes, like a pencil brush or a uniquely textured brush for cleanup. 

I'm not a fan of the basic vector brush with no texture lol

I'd love to hear your thoughts about the feature!

**Comment by Cat-Butler** [comment]
Note: As I understand it, this feature (available in a few apps), is based in the equivalent of ToonSquid's Path layers, and would require interaction between Path layers. 

ToonSquid's "vector brushes" are a completely different technology, for which this would not be appropriate.

_See also:_ [https://github.com/keiwando/toonsquid/discussions/39](https://github.com/keiwando/toonsquid/discussions/39 )

**Reply by ota-works** [reply]
Hopefully it could be reworked and implemented.

Fingers crossed
